mirascope/base/vectorstores.py

```diff
@@ -0,0 +1,30 @@
+from abc import ABC, abstractmethod
+from typing import Any, ClassVar, Optional, Union
+
+from pydantic import BaseModel
+
+from mirascope.base.chunkers import BaseChunker
+from mirascope.base.embedders import BaseEmbedder
+from mirascope.openai.embedders import OpenAIEmbedder
+
+from .types import BaseVectorStoreParams, Document
+
+
+class BaseVectorStore(BaseModel, ABC):
+    vectorstore_api_key: Optional[str] = None
+    index_name: ClassVar[Optional[str]] = None
+    chunker: ClassVar[BaseChunker] = BaseChunker()
+    embedder: ClassVar[Optional[BaseEmbedder]] = OpenAIEmbedder()
```

Comment:
this shouldn't rely on openai

Reply:
what should our base embedder be

Reply:
Just BaseEmbedder? This class can't be instantiated anyway since it has abstractmethod methods

Reply:
See `BaseCall` for reference